lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java

```diff
@@ -4004,7 +4006,19 @@ protected static boolean makeLink(UIContainer container, String ID, SimplePageIt
 
 		if (fake) {
 			ID = ID + "-fake";
-			UIOutput link = UIOutput.make(container, ID, i.getName());
+			String linkText = i.getName();
+			if (i.getType() == SimplePageItem.ASSIGNMENT) {
+				SecurityAdvisor yesMan = (String arg0, String arg1, String agr2) -> SecurityAdvisor.SecurityAdvice.ALLOWED;
+				securityService.pushAdvisor(yesMan);
+				try {
+					AssignmentEntity assignment = (AssignmentEntity) assignmentEntity.getEntity(i.getSakaiId(), simplePageBean);
+					linkText += " " + messageLocator.getMessage("simplepage.assignment.open_date", new Object[] {assignment.getOpenDate()});
+				} catch (Exception ex) {}
+				finally {
+					securityService.popAdvisor(yesMan);
+				}
+			}
```

Comment:
Perhaps there's a better way to do this in Lessons code? I'm not sure. I found this comment and code earlier in the file:

```
		    // assignments won't let us get the entity if we're not in the group, so set up permissions before other tests
			if (usable && i.isPrerequisite()) {
			    simplePageBean.checkItemPermissions(i, true);
			}
```

However if I call this method in the location here, nothing changes and I still get a permission exception in `AssignmentEntity` when trying to resolve the entity to access the open date.